### DISCUSSION TITLE
Added a file to installer. "Contact your support personnel"

### DISCUSSION THREAD

**Discussion by BryanCrotaz** [wixtoolset/issues#7835] [Questions]
Oh this is just ridiculous now. I added a dll to the component, and I got this error. It ran perfectly a minute or two earlier.
```xml
<Component Id="SystemManagement">
	<File
		BindPath="..\AgentService\bin\release\net7.0\publish"
		Source="System.Management.dll"/>
</Component>
```
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	WIX0001	WixToolset.Core.Native.Msi.MsiException (0x80004005): The Windows Installer service failed to start. Contact your support personnel	AgentSetupWix4	H:\Projects\Client2\Client\AgentSetupWix4\wix.exe	1	
```

Toolset errors should be actionable - ie it's fine to throw errors when I do something wrong, but the error must tell me what to do about it, or link to the docs that tell me what to do about it.

**Comment by BryanCrotaz** [comment]
https://learn.microsoft.com/en-us/windows/win32/msi/error-codes

Which one of that myriad of errors happened? Wix doesn't tell me.

**Comment by BryanCrotaz** [comment]
Rebooted, tried again, same problem straight away.

**Comment by robmen** [comment]
That's an exception. You can tell because the "dotted words" end with `Exception`. Exceptions in programming are called exceptions because they are [exceptional](https://www.bing.com/search?pglt=129&q=define+exceptional). In programming, "exceptional" is used as a synonym for "unusual" or "unexpected" not for "very good". Since exceptions are unexpected, they don't always provide the best error messages out of the box.

However, exceptions often do come with [stack traces](https://en.wikipedia.org/wiki/Stack_trace)! Stack traces can be very helpful in tracking down what failed unexpectedly. If you look around in more verbose output than what Visual Studio outputs by default, it is very likely you will find a stack trace.

Also, this is more of a general life tip but simply repeating the same behavior over and over without understanding what is causing the problem is not likely to make the problem go away. In fact, doing so is more likely to only make you more frustrated and possibly cause you to lash out at the very people who might just be able to help you.

I hope this message finds you better and that you learned a little bit today about exceptions, how they are unexpected and that stack traces can be very helpful to track down their root issue, and without understanding the root issue repeating the same behavior is unlikely to provide different results.

**Reply by BryanCrotaz** [reply]
I'm aware of what an exception is. My code has caused plenty of them! But if the exception code isn't surfaced into the log message they are incredibly hard to debug. 

Remember that this is happening in the build, not in the runtime so there is no stack trace. There's just what you choose to surface to the log. 

The bug here is that the error code from the underlying windows stuff isn't surfaced.
